docs: improve examples of `ndarray/iter` namespace

performant23 opened the pull request:
Updated examples to span over the functionalities implemented in the namespace.

Resolves #1589

## Description

> What is the purpose of this pull request?

Provide better examples for the namespace to demonstrate its capabilities and usage.

This pull request:
-   Imports different functionalities inside ndarray/iter to form useful usecases

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1589

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

performant23 committed:
docs: improve README examples of ndarray/iter namespace

Updated examples to span over the functionalities implemented in the namespace.

Addresses: #1589

performant23 commented:
Hello, thought I should just push a draft PR for one namespace and receive feedback before moving to another in the package in ndarray! 
Also, I have these questions:
1. Do we include the outputs of the examples as comments for README.md?
2. Do we need to format the output in some particular way? That is, in the second example, the log is like: Index: [ 0, 0 ] Value: 1, and for the last one, we print the initial type of the object like: Original ndarray: ndarray( 'generic', [ 1, 2, 3, 4, 5, 6, 7, 8, 9 ], [ 3, 3 ], [ 3, 1 ], 0, 'row-major' ). So, is that necessary or we should just always log the final output?

Thank you so much!!

performant23 committed:
Update README.md

Update the copyright year

Signed-off-by: Rutam <138517416+performant23@users.noreply.github.com>

kgryte commented on lib/node_modules/@stdlib/ndarray/iter/README.md:
```diff
@@ -71,12 +71,151 @@ var o = ns;
 var objectKeys = require( '@stdlib/utils/keys' );
 var ns = require( '@stdlib/ndarray/iter' );
 
 console.log( objectKeys( ns ) );
```
Prefer modifying the existing example code. Project conventions are to not create multiple separate Markdown code blocks. Prefer instead a single Markdown code block with various example usage.

kgryte commented on lib/node_modules/@stdlib/ndarray/iter/README.md:
```diff
@@ -71,12 +71,151 @@ var o = ns;
+var iterEntries = ns.nditerEntries( x );
+var entry;
+
+for ( entry of iterEntries ) {
```
ES5.

kgryte commented on lib/node_modules/@stdlib/ndarray/iter/README.md:
````diff
@@ -71,12 +71,151 @@ var o = ns;
+```
+
+```javascript
+// Example: Iterate over each row of the ndarray
````
These examples seem to just be pulling from the READMEs of the child packages. I think we would prefer something fresh, which ideally combines the functionality of multiple packages to demonstrate how they can all work together.

performant23 committed:
docs: apply suggestions from code review

Updated example for the namespace using functionalities from stats, random, and ndarray

Fixes: #1589

performant23 committed:
Merge remote-tracking branch 'origin/docs/improve_README_examples_of_ndarray/iter_namespace' into docs/improve_README_examples_of_ndarray/iter_namespace

performant23 commented:
Thank you so much for the review, @kgryte! I have modified the commit to include an example which uses functionalities from other packages! Please do let me know if there are any further changes necessary. 

Also, I found that packages followed the convention of including comments below code. Since I was getting the below message, I left a line between the code and the necessary comments. Please do let me know in case we need to include corresponding comments above the code.
```
Missing empty line before comment  stdlib/empty-line-before-comment
```

performant23 committed:
Merge branch 'stdlib-js:develop' into docs/improve_README_examples_of_ndarray/iter_namespace

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into docs/improve_README_examples_of_ndarray/iter_namespace
